hack/ci/e2e-conformance.sh

```diff
@@ -25,47 +25,143 @@ CLUSTER_NAME=${CLUSTER_NAME:-"test1"}
+
   # remove our tempdir
   # NOTE: this needs to be last, or it will prevent kind delete
   if [[ -n "${TMP_DIR:-}" ]]; then
-    rm -rf "${TMP_DIR}"
+    rm -rf "${TMP_DIR}" || true
   fi
 }
 
+# SSH to a node by name ($1) and run a command ($2).
+function ssh-to-node() {
+  local node="$1"
+  local zone="$2"
+  local cmd="$3"
+
+  # ensure we have an IP to connect to
+  gcloud compute --project "${GCP_PROJECT}" instances add-access-config "${node}" || true
+
+  # Loop until we can successfully ssh into the box
+  for try in {1..5}; do
+    if gcloud compute ssh --ssh-flag="-o LogLevel=quiet" --ssh-flag="-o ConnectTimeout=30" \
+      --project "${GCP_PROJECT}" --zone "${zone}" "${node}" --command "echo test > /dev/null"; then
+      break
+    fi
+    sleep 5
+  done
+  # Then actually try the command.
+  gcloud compute ssh --ssh-flag="-o LogLevel=quiet" --ssh-flag="-o ConnectTimeout=30" \
+    --project "${GCP_PROJECT}" --zone "${zone}" "${node}" --command "${cmd}"
+}
+
 init_image() {
   image=$(gcloud compute images list --project $GCP_PROJECT \
     --no-standard-images --filter="family:capi-ubuntu-1804-k8s" --format="table[no-heading](name)")
   if [[ -z "$image" ]]; then
+      if ! command -v ansible &> /dev/null; then
+        if [[ $EUID -ne 0 ]]; then
+          echo "Please install ansible and try again."
+          exit 1
+        else
+          # we need pip to install ansible
+          curl -L https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+          python get-pip.py --user
+          rm -f get-pip.py
+
+          # install ansible needed by packer
+          version="2.8.5"
+          python -m pip install "ansible==${version}"
+        fi
+      fi
       if ! command -v packer &> /dev/null; then
         hostos=$(go env GOHOSTOS)
         hostarch=$(go env GOHOSTARCH)
         version="1.4.3"
         url="https://releases.hashicorp.com/packer/${version}/packer_${version}_${hostos}_${hostarch}.zip"
         echo "Downloading packer from $url"
-        wget -O packer.zip $url  && \
+        wget --quiet -O packer.zip $url  && \
           unzip packer.zip && \
           rm packer.zip && \
-          mv packer "$(go env GOPATH)/bin"
+          ln -s $PWD/packer /usr/local/bin/packer
+      fi
+      if [[ $EUID -ne 0 ]]; then
+        (cd "$(go env GOPATH)/src/sigs.k8s.io/image-builder/images/capi" && \
+          GCP_PROJECT_ID=$GCP_PROJECT GOOGLE_APPLICATION_CREDENTIALS=$GOOGLE_APPLICATION_CREDENTIALS \
+          make build-gce-default)
+      else
+        # assume we are running in the CI environment as root
+        # Add a user for ansible to work properly
+        groupadd -r packer && useradd -m -s /bin/bash -r -g packer packer
+        # use the packer user to run the build
+        su - packer -c "bash -c 'cd /go/src/sigs.k8s.io/image-builder/images/capi && GCP_PROJECT_ID=$GCP_PROJECT GOOGLE_APPLICATION_CREDENTIALS=$GOOGLE_APPLICATION_CREDENTIALS make build-gce-default'"
```

Comment:
I'm wondering if we need to add support to the kubeadm provider to override binaries used, then we wouldn't need to re-bake an image just to run the tests.

Reply:
sounds like a good idea!